packages/core/src/app/checkout/Checkout.tsx

```diff
@@ -500,22 +518,29 @@ class Checkout extends Component<CheckoutProps & WithCheckoutProps & WithLanguag
         }
 
         const nextStepIndex = findIndex(steps, { type })
-        const prevStepIndex = nextStepIndex - 1;
-        const prevStep = prevStepIndex >= 0 && steps[prevStepIndex];
+        let prevStepIndex = nextStepIndex - 1;
+        let prevStep = prevStepIndex >= 0 && steps[prevStepIndex];
 
         /**
          * BOLT ANALYTICS
          * This block is to handle emitting step completion events when details are autocompleted.
          * Events will still be emitted from the individual steps and will set completion state flags
          * so that when this block runs during step transitions, we don't emit duplicate events.
          */
+        // skip previous steps that aren't required (e.g. shipping is not required for carts with only digital products)
+        while (prevStepIndex > 0 && prevStep && !prevStep.isRequired) {
+            prevStep = steps[--prevStepIndex];
+        }
```

Comment:
Is this only needed for digital products? If so, I don't think we should do this because we're deviating too far away from the main checkout-js repo. LAPG also doesn't have digital products

Reply:
yeah this is mainly to cover any similar cases to digital products